src/formats/notion-api.ts

```diff
@@ -0,0 +1,1147 @@
+					coverPropertyName: this.coverPropertyName,
+					// Callback to import database pages
+					importPageCallback: async (pageId: string, parentPath: string, databaseTag?: string) => {
+						await this.fetchAndImportPage(ctx, pageId, parentPath, databaseTag);
+					},
+					onPagesDiscovered: (newPagesCount: number) => {
+						// Callback provided but not used - progress is reported per page/attachment
+					}
+				}
+			);
+			
+			// Clear the cache after processing this page to free memory
+			blocksCache.clear();
+			
+			// Prepare YAML frontmatter
+			// Start with notion-id and notion-db at the top
+			const frontMatter: Record<string, any> = {
+				'notion-id': page.id,
+			};
+		
+			// Add database tag if this page belongs to a database (right after notion-id)
+			if (databaseTag) {
+				frontMatter['notion-db'] = databaseTag;
+			}
+		
+			// Extract all other properties from the page
+			const extractedProps = extractFrontMatter(page, this.formulaStrategy);
+			// Merge extracted properties (skip notion-id as we already added it)
+			for (const key in extractedProps) {
+				if (key !== 'notion-id') {
+					frontMatter[key] = extractedProps[key];
+				}
+			}
+		
+			// Process cover image if present
+			if (frontMatter.cover && typeof frontMatter.cover === 'string') {
+				try {
+				// Determine cover type based on URL
+					const coverUrl = frontMatter.cover;
+					const isExternal = !coverUrl.includes('secure.notion-static.com');
+				
+					// Cover images are always downloaded, regardless of downloadExternalAttachments setting
+					// This is because Notion covers often use external URLs even for Notion-hosted images
+					// Use the page title as the cover filename for better organization
+					const result = await downloadAttachment(
+						{
+							type: isExternal ? 'external' : 'file',
+							url: coverUrl,
+							name: sanitizedTitle // Use page title as cover filename
+						},
+						{
+							ctx,
+							currentFolderPath: pageFolderPath,
+							client: this.notionClient!,
+							vault: this.vault,
+							downloadExternalAttachments: true, // Always download cover images
+							currentPageTitle: sanitizedTitle
+						}
+					);
+		
+					// For frontmatter, use wiki link syntax with double quotes for proper rendering
+					// Cover images should always be downloaded locally
+					if (result.isLocal && result.filename) {
+						// Report progress for cover image download
+						this.attachmentsDownloaded++;
+						ctx.attachments = this.attachmentsDownloaded;
+						ctx.attachmentCountEl.setText(this.attachmentsDownloaded.toString());
+					
+						// Extract extension from filename
+						const ext = result.filename.substring(result.filename.lastIndexOf('.'));
+						const fullPath = result.path + ext;
+						// Use wiki link syntax with double quotes: "[[path]]"
+						// The double quotes are necessary for YAML to render it as a link
+						const coverValue = `[[${fullPath}]]`;
```

Comment:
It looks like the path for the cover image is incorrect. Clicking on it shows a notice "Folder already exists". This path should probably be created using `app.fileManager.generateMarkdownLink`

<img width="1028" height="517" alt="image" src="https://github.com/user-attachments/assets/62746834-cb5c-4061-9719-e80e0b3cf5da" />

Reply:
I'm noticing this problem in all places where references are made to attachments.

Reply:
The font in the screenshot suggests you’re using Windows, and your link format is `Relative to file`, right?
I haven’t tested Windows yet, and my option is `Shortest path when possible`. I’m not sure if either of these is affecting the issue, because I use the normalizePath function everywhere a path appears. I happen to have a Windows computer, so I’ll test it today to see what’s going on. The screenshot below shows my Mac attachment path:
<img width="1572" height="412" alt="CleanShot_2025_11_13_at_09_40_41@2x" src="https://github.com/user-attachments/assets/ba1a7500-afe0-4e44-a4e0-9cbf702ca545" />
<img width="1456" height="358" alt="CleanShot_2025_11_13_at_09_42_17@2x" src="https://github.com/user-attachments/assets/512a31cf-95b6-4b43-bf41-b25f9c07a421" />